impeller/renderer/backend/vulkan/allocator_vk.cc

```diff
@@ -132,12 +132,27 @@ static constexpr vk::ImageUsageFlags ToVKImageUsageFlags(PixelFormat format,
 
   if (usage & static_cast<TextureUsageMask>(TextureUsage::kShaderRead)) {
     vk_usage |= vk::ImageUsageFlagBits::eSampled;
+    // unmark the usage as transient if it is used as a sampled image
+    // See: https://github.com/flutter/flutter/issues/121633
+    if (mode == StorageMode::kDeviceTransient) {
```

Comment:
This looks like a straight up bug in Impeller where are setting the usage of the texture to be shader read but using a device transient storage mode. Can we get a backtrace of how this is happening?